config/admin-functions/generate-key.sh

```diff
@@ -16,13 +23,17 @@ tpm2_createprimary -C o -c primary.ctx
 tpm2_evictcontrol -Q -c 0x81000000 || true 2>&1 /dev/null
 tpm2_evictcontrol -c primary.ctx 0x81000000
 
-# Create a policy
-tpm2_create -L pcr.policy -u key.pub -C primary.ctx -G ecc -c key.ctx
+# Create keys
+tpm2_create -L pcr.policy -u key.pub -r key.priv -C primary.ctx -G ecc 
+tpm2_load -u key.pub -r key.priv -C primary.ctx -c key.ctx
 
 # Save the keys
 tpm2_evictcontrol -Q -c 0x81000001 || true 2>&1 /dev/null
 tpm2_evictcontrol -c key.ctx 0x81000001
 
+# Delete the local key files
+rm -rf key.pub key.priv
```

Comment:
should we `shred` the private key for extra safety?

Reply:
We can, it is already encrypted. What do you think?

Reply:
Yeah I'm not terribly worried, just doing defense in depth you know, in case later we screw up some setting and it doesn't encrypt the private key (why it would do that.. who knows.) Just `shred -u key.priv` is an easy little extra layer of defense.

Reply:
I'm adding a comment explaining that `key.priv` is encrypted with a key that can't be extracted from the TPM.